### ISSUE TITLE
Custom Script Animation (Don't show prev screen)

### ISSUE THREAD

**Issue by Avatarchik** [grashaar/uiman#1]
Hi! When opening a new screen, the stryy does not hide when using custom animation

**Comment by grashaar** [comment]
Hi! Sorry for my very late reply. When you use custom script animation you have to override these 2 methods and provide your own logic to show and hide the UI.

```
public override IEnumerator AnimationHide() { }
public override IEnumerator AnimationShow() { }
```

**Comment by Avatarchik** [comment]
Hi! Hide animation (current screen) does not work when switching to another screen

I am doing this kind of work, but can you show me how to solve it more correctly

```
private IEnumerator DelayDequeueDialog(IEnumerator coroutine, UIManBase ui, bool resetDialogTransitionStatus,bool hide)
{
            yield return StartCoroutine(coroutine);

            this.IsInDialogTransition = false;

            if (hide)
            {
                OnHideUIComplete(ui);
                ui.OnHideComplete();
            }
            else
            {
                ui.UnlockInput();
                ui.OnHideComplete();
                OnShowUIComplete(ui);
            }
          

            if (ui.GetUIBaseType() == UIBaseType.Dialog && !resetDialogTransitionStatus)
                DequeueDialog();
 }
```

**Comment by grashaar** [comment]
You are right, custom script animations are not working correctly. I've fixed it in 1.4.30. Please take a look at the updated example (https://github.com/grashaar/uiman-examples/commit/03c765d7b53fc9fbf577bd3818dde288d7782261).
